src/Mvc/Mvc.Razor/src/RazorPageBase.cs

```diff
@@ -644,9 +644,11 @@ public virtual async Task<HtmlString> FlushAsync()
         public virtual HtmlString SetAntiforgeryCookieAndHeader()
         {
             var viewContext = ViewContext;
-            var antiforgery = viewContext?.HttpContext.RequestServices.GetRequiredService<IAntiforgery>();
-            antiforgery.SetCookieTokenAndHeader(viewContext?.HttpContext);
-
+            if (viewContext != null)
+            {
+                var antiforgery = viewContext.HttpContext.RequestServices.GetRequiredService<IAntiforgery>();
+                antiforgery.SetCookieTokenAndHeader(viewContext.HttpContext);
+            }
```

Comment:
@pranavkm - do you know why this method tolerates null? Is this for unit testing?

Reply:
Fairly certain it's due to unit testing. I think it would be fair to not do null-checks here. (Same with the `FlushAsync` above). We would never expect the ViewContext to be null here when the page is executing.

Reply:
I think we should just take this change then. It doesn't seem overly important which semantic we support